monkey/tests/unit_tests/agent_plugins/exploiters/powershell/test_credentials_generator.py

```diff
@@ -0,0 +1,183 @@
+
+    if running_on_windows:
+        assert Credentials(identity=LOCAL_USER_IDENTITY, secret=None) in credentials
+
+
+def assert_empty_password_credentials_in_results(
+    credentials: Sequence[Credentials], running_on_windows: bool
+):
+    for i in INPUT_IDENTITIES:
+        assert Credentials(identity=i, secret=Password(password="")) in credentials
+
+    if running_on_windows:
+        assert (
+            Credentials(identity=LOCAL_USER_IDENTITY, secret=Password(password="")) in credentials
+        )
+
+
+def assert_correct_identity_types(credentials: Sequence[Credentials]):
+    for c in credentials:
+        assert isinstance(c.identity, Username) or c.identity is None
+
+
+def assert_correct_secret_types(credentials: Sequence[Credentials]):
+    for c in credentials:
+        assert type(c.secret) in SECRET_TYPES or c.secret is None
+
+
+def assert_domain_users_added(credentials: Sequence[Credentials]):
+    for c in credentials:
+        if c.identity is None:
+            continue
+
+        if "\\" not in c.identity.username:
+            assert (
+                Credentials(
+                    identity=Username(username=f"{TEST_DOMAIN}\\{c.identity.username}"),
+                    secret=c.secret,
+                )
+                in credentials
+            )
+
+
+def assert_local_domain_usernames_added(credentials: Sequence[Credentials]):
+    for c in credentials:
+        if c.identity is None:
+            continue
+
+        if "\\" not in c.identity.username:
+            assert (
+                Credentials(
+                    identity=Username(username=f".\\{c.identity.username}"), secret=c.secret
+                )
+                in credentials
+            )
+
+
+def test_empty_windows_true():
+    generated_credentials = generate_powershell_credentials([], True)
+
+    assert_cached_credentials_in_results(generated_credentials)
+
+
+def test_empty_windows_false():
+    generated_credentials = generate_powershell_credentials([], False)
+
+    assert len(generated_credentials) == 0
+
+
+def test_usernames_only_windows_true():
+    input_credentials = [Credentials(identity=i, secret=None) for i in INPUT_IDENTITIES]
+    generated_credentials = generate_powershell_credentials(input_credentials, True)
+
+    assert_cached_credentials_in_results(generated_credentials)
+    assert_cached_password_credentials_in_results(generated_credentials, True)
+    assert_empty_password_credentials_in_results(generated_credentials, True)
+    assert_domain_users_added(generated_credentials)
+    assert_local_domain_usernames_added(generated_credentials)
+    assert len(generated_credentials) == 19
+
+
+def test_usernames_only_windows_false():
+    input_credentials = [Credentials(identity=i, secret=None) for i in INPUT_IDENTITIES]
+    generated_credentials = generate_powershell_credentials(input_credentials, False)
+
+    assert len(generated_credentials) == 4
```

Comment:
It might be cool to be able to assert that, say, the cached credentials are *not* part of the generated credentials